### PR TITLE
Run simple build on Mesos Plugin in integration test.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This patch introduces a simple integration test that runs a build using the Mesos plugin.
The mayor part is the introduction of a Jenkins configuration client that sets all form fields
according the the Jenkins UI.

JIRA issues: DCOS_OSS-4890

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,9 +22,12 @@ dependencies {
     api ('com.mesosphere.usi:core') { version { branch = usiBranch } }
     api ('com.mesosphere.usi:core-models') { version { branch = usiBranch } }
     api ('com.mesosphere.usi:mesos-client') { version { branch = usiBranch } }
+    api 'com.squareup.okhttp3:okhttp:3.14.1'
```
Comment (jeschkies) [inline build.gradle]: If someone want to replace the `OkHttpClient` with Akka HTTP feel free to do so. I'm out for the next week 🙂

Comment (timcharper) [reply build.gradle]: nah premature optimization; I'm sure a blocking client is just fine for Jenkins

Comment (timcharper) [reply build.gradle]: oh and it's test code LOL so NO

Comment (kvish) [reply build.gradle]: yup agreed this is fine for testing,

File: src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
+        .add("administrativeMonitor", "on")
```
Comment (zen-dog) [inline src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java:245]: 😂 😂 😂 so much magic here, it just blows my mind

File: src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+  public Response addMesosCloud(
+      String mesosMasterUrl,
+      String frameworkName,
+      String role,
+      String agentUser,
+      String jenkinsUrl,
+      String label,
+      String mode)
+      throws IOException, UnsupportedEncodingException {
+    final String jsonData =
```
Comment (zen-dog) [inline src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java:56]: Do you want to leave a json example of how it should look like at the end as a comment?

**Comment by kvish** [comment]
overall lgtm we can improve this test as we iterate

**Review by kvish — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        .add(
+            "jenkins-model-JenkinsLocationConfiguration",
+            Json.createObjectBuilder().add("url", jenkinsUrl).add("adminAddress", "").build())
+        .add("hudson-task-Shell", Json.createObjectBuilder().add("shell", "").build());
+  }
+
+  private FormDataBuilder addFormDefaults(FormDataBuilder builder, String jenkinsUrl)
+      throws UnsupportedEncodingException {
+    return builder
+        .add("system_message", "")
```
Comment (zen-dog) [inline src/test/java/org/jenkinsci/plugins/mesos/JenkinsConfigClient.java:204]: How we ever going to maintain this class after Jenkins updates is beyond me 😭

**Review by zen-dog — APPROVED** [review]
I'm just going to pretend that I never saw the `JenkinsConfigClient.java` 🙈 Don't hold this approve against me!

**Pull request merged**